qa-tests-backend/src/test/groovy/ImageSignatureVerificationTest.groovy

```diff
@@ -206,7 +204,11 @@ w9e2Azq1OYIh/pbeBMHARDrBaqqmuMR9+BfAaPAYdkNTU6f58M2zBbuL0A==
     def "Check violations of policy '#policyName' for deployment '#deployment.name'"() {
         expect:
         "Verify deployment has expected violations"
-        assert waitForViolation(deployment.name, policyName, WAIT_FOR_VIOLATION_TIMEOUT) == expectViolations
+        if (expectViolations) {
+            assert waitForViolation(deployment.name, policyName, 15) == expectViolations
+        } else {
+            assert checkForNoViolations(deployment.name, policyName, 15)
```

Comment:
This makes me happier but not 100% happy. As I was suggesting previously, let's have timeout for `checkForNoViolations` N (2,3,...) times bigger than the timeout for `waitForViolation`. This will tell the reader: _Look, we waited **more than enough** for violation to appear but it had not appeared._
```suggestion
            assert waitForViolation(deployment.name, policyName, 5) == expectViolations
        } else {
            assert checkForNoViolations(deployment.name, policyName, 15)
```